opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/MLCommonsOperator.java

```diff
@@ -91,15 +96,24 @@ public ExprValue next() {
     return Collections.singletonList(input);
   }
 
-  protected MLAlgoParams convertArgumentToMLParameter(Argument argument, String algorithm) {
+  protected MLAlgoParams convertArgumentToMLParameter(Map<String, Literal> arguments,
+                                                      String algorithm) {
     switch (FunctionName.valueOf(algorithm.toUpperCase())) {
       case KMEANS:
-        if (argument.getValue().getValue() instanceof Number) {
-          return KMeansParams.builder().centroids((Integer) argument.getValue().getValue()).build();
-        } else {
-          throw new IllegalArgumentException("unsupported Kmeans argument type:"
-                  + argument.getValue().getType());
-        }
+        return KMeansParams.builder()
+                .centroids(arguments.containsKey(CENTROIDS)
+                        ? ((Integer) arguments.get(CENTROIDS).getValue())
```

Comment:
just wondering if type check is done somewhere like in Analyzer?

Reply:
Parameter type is defined in antlr parser, wrong parameter type will be rejected at antlr parser level.
Not going to add parsing/checking logical in Analyzer because I'm trying to avoid adding ml-commons dependency in core package.

Reply:
got it. I just saw it in parser.g4. thanks